### ISSUE TITLE
ggttggg madevent/fortran segfaults on juwels (common blocks too large, workaround is reducing NB_PAGE_MAX from 16k to 32)

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#498]
I am running on juwels cluster login nodes again to see the x8 of AVX512 on Intel Gold.

I see it nicely for ggttgg, But ggttggg segfaults
```
DATE: 2022-06-24_11:22:55

On jwlogin05.juwels [CPU: Intel(R) Xeon(R) Gold 6148 CPU] [GPU: none]:
Working directory (run): /p/home/jusers/valassi1/juwels/madgraph4gpu/epochX/cudacpp/gg_ttggg.mad/SubProcesses/P1_gg_ttxggg

*** (1) EXECUTE MADEVENT (create results.dat) ***
--------------------
64 1 1 ! Number of events and max and min iterations
0.000001 ! Accuracy (ignored because max iterations = min iterations)
0 ! Grid Adjustment 0=none, 2=adjust (NB if = 0, ftn26 will still be used if present)
1 ! Suppress Amplitude 1=yes (i.e. use MadEvent single-diagram enhancement)
0 ! Helicity Sum/event 0=exact
1 ! Channel number (1-N) for single-diagram enhancement multi-channel (NB used even if suppress amplitude is 0!)
--------------------
Executing ' ./madevent < /tmp/valassi1/input_ggttggg_x1_fortran > /tmp/valassi1/output_ggttggg_x1_fortran'

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x14f9391f33ff in ???
#1  0x45c160 in ???
#2  0x401eae in ???
#3  0x14f9391df492 in ???
#4  0x401eed in ???
#5  0xffffffffffffffff in ???
./madX.sh: line 266: 10157 Segmentation fault      $timecmd $cmd < ${tmpin} > ${tmp}
ERROR! ' ./madevent < /tmp/valassi1/input_ggttggg_x1_fortran > /tmp/valassi1/output_ggttggg_x1_fortran' failed

```

**Comment by valassi** [comment]
Funnily enough, this happens immediately on madevent, without the need for any arguments.

Rebuilding with -g, gdb seems to say this happens in driver.f at NB_PAGE_LOOP=32 !

I then tried to use valgrind on this madevent, I get some
```
valgrind: mmap(0x877000, 3613958144) failed in UME with error 22 (Invalid argument).
valgrind: this can be caused by executables with very large text, data or bss segments.
```

There are many pointers to this on the web, but I cannot find an easy solution. One explicitly mentioning fortran is
https://userinfo.surfsara.nl/systems/shared/software/valgrind
```
if your code contains common blocks with large, pre-defined arrays, you can use the Intel compiler option -dyncom "common1,common2,.." to resolve this issue.
```

This last pointer was useful not in itself, but because it shows the right direction: in ggttggg the commons are too large.

I have added a very simple workaround, only for a one-off test in juwels: I have changed NB_PAGE_MAX from 16384 to 32. This is ok anyway on Juwles cluster as I really only want to test C++, not CUDA. So I do not need the 16k events in flight of the GPU. This seems to work.

I will keep this open anyway, we need a solution, this was just a worakround...
